public/controllers/register-agent/containers/register-agent/register-agent.tsx

```diff
@@ -0,0 +1,240 @@
+    };
+
+    const form = useForm(initialFields);
+
+    const getRemoteConfig = async () => {
+      const remoteConfig = await getMasterRemoteConfiguration();
+      if (remoteConfig) {
+        setHaveUdpProtocol(remoteConfig.isUdp);
+        setHaveConnectionSecure(remoteConfig.haveSecureConnection);
+        setUdpProtocol(remoteConfig.isUdp);
+        setConnectionSecure(remoteConfig.haveSecureConnection);
+      }
+    };
+
+    const getAuthInfo = async () => {
+      try {
+        const result = await WzRequest.apiReq(
+          'GET',
+          '/agents/000/config/auth/auth',
+          {},
+        );
+        return (result.data || {}).data || {};
+      } catch (error) {
+        ErrorHandler.handleError(error);
+      }
+    };
+
+    useEffect(() => {
+      const fetchData = async () => {
+        try {
+          const wazuhVersion = await getWazuhVersion();
+          let wazuhPassword = '';
+          let hidePasswordInput = false;
+          await getRemoteConfig();
+          const authInfo = await getAuthInfo();
+          const needsPassword = (authInfo.auth || {}).use_password === 'yes';
+          if (needsPassword) {
+            wazuhPassword =
+              configuration['enrollment.password'] ||
+              authInfo['authd.pass'] ||
+              '';
+            if (wazuhPassword) {
+              hidePasswordInput = true;
+            }
+          }
+          const groups = await getGroups();
+
+          setNeedsPassword(needsPassword);
+          setHidePasswordInput(hidePasswordInput);
+          setWazuhPassword(wazuhPassword);
+          setWazuhVersion(wazuhVersion);
+          setGroups(groups);
+          setLoading(false);
+        } catch (error) {
+          setWazuhVersion(wazuhVersion);
+          setLoading(false);
+          const options = {
+            context: 'RegisterAgent',
+            level: UI_LOGGER_LEVELS.ERROR,
+            severity: UI_ERROR_SEVERITIES.BUSINESS,
+            display: true,
+            store: false,
+            error: {
+              error: error,
+              message: error.message || error,
+              title: error.name || error,
+            },
+          };
+          ErrorHandler.handleError(error, options);
+        }
+      };
+
+      fetchData();
+    }, []);
+
+    const agentGroup = <InputForm {...form.fields.agentGroups}></InputForm>;
+    const osCard = (
+      <InputForm {...form.fields.operatingSystemSelection}></InputForm>
+    );
+
+    return (
+      <div>
+        <div className='close'>
+          {hasAgents() && (
+            <EuiButtonEmpty
+              size='s'
+              onClick={() => addNewAgent(false)}
+              iconType='cross'
+            ></EuiButtonEmpty>
+          )}
+          {!hasAgents() && (
+            <EuiButtonEmpty
+              size='s'
+              onClick={() => reload()}
+              iconType='refresh'
+            >
+              Refresh
+            </EuiButtonEmpty>
+          )}
```

Comment:
**suggestion**: the `hasAgents` function is executed 2 times, you could save the result to a variable and use it instead.

**suggestion**: for this case, you could use a ternary to render the button